### PR TITLE
Automation test for Wiki aliases

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Automation test for Wiki aliases 

#### Related Pull Requests
https://github.com/LabKey/platform/pull/3165

#### Changes
1. Manage configuration page : ManageWikiConfigurationPage.java
2. New test class : WikiAliasesTest.java

### CODE REVIEW

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        log("Verifying  all the aliases added works with wiki-page.view");
+        manageWikiConfigurationPage = wikiHelper.manageWikiConfiguration();
+        String listOfAliases[] = manageWikiConfigurationPage.getAliases().split("\n");
+        for (int i = 0; i < listOfAliases.length; i++)
+        {
+            beginAt(WebTestHelper.buildURL("wiki", getProjectName(), "page", Map.of("name", listOfAliases[i])));
+            Assert.assertEquals("Incorrect wiki body for alias " + listOfAliases[i], wikiBody,
+                    Locator.tagWithClass("div", "labkey-wiki").findElement(getDriver()).getText());
+        }
+
```
Comment (labkey-adam) [inline src/org/labkey/test/tests/wiki/WikiAliasesTest.java:97]: Would be good to test that using aliases is case-insensitive. Here's one way:

`String listOfAliases[] = (manageWikiConfigurationPage.getAliases() + "\n" + manageWikiConfigurationPage.getAliases().toLowerCase()).split("\n");`

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        String duplicateAlias = "Alias5";
+        String aliases = "Alias1\n" + "Alias2\n" + "Alias3\n" + "Alias4\n" + duplicateAlias;
+        wikiHelper.createWikiPage(wikiName, null, wikiTitle, wikiBody, true, null, false);
+
+        log("Verifying adding new alias with rename button");
+        ManageWikiConfigurationPage manageWikiConfigurationPage = wikiHelper.manageWikiConfiguration();
+        manageWikiConfigurationPage.rename(duplicateAlias, true)
+                .setAliases(aliases)
+                .save();
+
```
Comment (labkey-adam) [inline src/org/labkey/test/tests/wiki/WikiAliasesTest.java:87]: There's nothing that verifies that an alias was automatically created for `wikiName`. I would first do a rename **without** setting aliases and verify that `wikiName` still works. Then add more aliases and verify those.

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            Assert.assertEquals("Incorrect wiki body for alias " + listOfAliases[i], wikiBody,
+                    Locator.tagWithClass("div", "labkey-wiki").findElement(getDriver()).getText());
+        }
+
+        log("Verifying same alias name is not allowed");
+        manageWikiConfigurationPage = wikiHelper.manageWikiConfiguration();
+        String errMsg = manageWikiConfigurationPage.setAliases( "\n" + duplicateAlias)
+                .saveExpectingErrors();
+        Assert.assertEquals("Incorrect error message", "Warning: Alias '" + duplicateAlias + "' already exists in this folder.",errMsg);
+
```
Comment (labkey-adam) [inline src/org/labkey/test/tests/wiki/WikiAliasesTest.java:103]: Probably most interesting to verify that adding a duplicate alias on a different wiki fails to add that alias and produces an error

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        log("Testing case insensitive");
+        manageWikiConfigurationPage = wikiHelper.manageWikiConfiguration();
+        errMsg = manageWikiConfigurationPage.setAliases("\n" + duplicateAlias.toLowerCase()).saveExpectingErrors();
+        Assert.assertEquals("Incorrect error message", "Warning: Alias '" + duplicateAlias.toLowerCase() + "' already exists in this folder.",errMsg);
+
```
Comment (labkey-adam) [inline src/org/labkey/test/tests/wiki/WikiAliasesTest.java:120]: A good additional test would be to create a new wiki with a name that matches one of the aliases (you could use `wikiName`) and verify that page.view resolves to the wiki with that name, not the wiki that has that alias.

**Review by labkey-adam — CHANGES_REQUESTED** [review]
A few suggestions

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java (hunk 2 above)
Comment (labkey-adam) [inline src/org/labkey/test/tests/wiki/WikiAliasesTest.java:87]: Should also verify that the rename worked (before adding aliases)

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java (hunk 1 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/wiki/WikiAliasesTest.java:97]: Done

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java (hunk 3 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/wiki/WikiAliasesTest.java:103]: Done

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java (hunk 2 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/wiki/WikiAliasesTest.java:87]: Done

File: src/org/labkey/test/tests/wiki/WikiAliasesTest.java (hunk 4 above)
Comment (labkey-sweta) [reply src/org/labkey/test/tests/wiki/WikiAliasesTest.java:120]: Good suggestion

**Review by labkey-adam — APPROVED** [review]

**Pull request merged**